File: Javascript/Algoritm/Sorting/countingSort2.js

function countingSort(arr) {
    let sortMap = new Array(100).fill(0);

    for(let i in arr) {
        sortMap[[arr[i]]] += 1;
    }

    let result = new Array()
    //print
    for(let i in sortMap) {
        for(let j=0; j<sortMap[i]; j++) {
            result.push(i);
        }
    }
    return result;
}

var input = [19, 10, 12, 10, 24, 25, 22];
countingSort(input);

File: C++/Algoritm/Sorting/countingSort2.cpp
#include <iostream>
#include <vector>

using namespace std;
#define MAX_ARR 4

vector<int> countingSort(vector<int> arr)
{   
    vector<int> countArr;
    vector<int> output;
    countArr.assign(MAX_ARR, 0);
    
    for(int i=0; i < arr.size(); i++)
    {
        int index = arr[i];
        countArr[index] += 1;
    }
    
    for(int i=0; i<MAX_ARR; i++)
    {
        while(countArr[i] != 0)
        {
            cout << i << " ";
            countArr[i] -= 1;
        }
    }

    return countArr;
}

int main()
{
    vector<int> mArray = {1, 1, 3, 2, 1};
    vector<int> result = countingSort(mArray);

    return 0;
}